app/code/community/Bolt/Boltpay/controllers/ShippingController.php

```diff
@@ -343,35 +283,23 @@ public function getPrefetchCacheIdentifier($quote, $addressData)
             $cacheIdentifier .= '_' . $addressData['postcode'];
         }
 
+        if (isset($addressData['city'])) {
+            $cacheIdentifier .= '_' . $addressData['city'];
+        }
+
+        if (isset($addressData['region'])) {
+            $cacheIdentifier .= '_' . $addressData['region'];
+        }
+
+        if (isset($addressData['region_id'])) {
+            $cacheIdentifier .= '_' . $addressData['region_id'];
+        }
+
         // include products in cache key
         foreach($quote->getAllVisibleItems() as $item) {
             $cacheIdentifier .= '_'.$item->getProductId().'_'.$item->getQty();
         }
 
         return md5($cacheIdentifier);
     }
```

Comment:
To cover cases when shipping may differ after applying discounts / gift cards quote applied rules (discounts) and perhaps getTotals response array keys should be taken into account when generating cache identifier.